File: fractal/src/main/java/net/fwitz/math/fractal/escape/collatz/Collatz.java
package net.fwitz.math.fractal.escape.collatz;

import net.fwitz.math.binary.complex.Complex;

public class Collatz {
    public static Complex collatz(Complex z) {
        Complex onePlus4z = z.times(4).plus(1);
        Complex onePlus2z = z.times(2).plus(1);
        Complex cosPiZ = z.times(Math.PI).cos();
        return onePlus4z.minus(onePlus2z.times(cosPiZ)).times(0.25);
    }
}

File: fractal/src/main/kotlin/net/fwitz/math/fractal/escape/collatz/Collatz.kt
package net.fwitz.math.fractal.escape.collatz

import net.fwitz.math.binary.complex.Complex

object Collatz {
    fun collatz(z: Complex): Complex {
        val onePlus4z = z * 4 + 1
        val onePlus2z = z * 2 + 1
        val cosPiZ: Complex = (z * Math.PI).cos
        return (onePlus4z - (onePlus2z * cosPiZ)) * 0.25
    }
}